### PR TITLE
Implemented feedbacks from previous PR - backend (seatbid)

### PR DESCRIPTION
**Pull request by nowshin-108**
### Key Changes:

1. Implemented seat ownership verification in `/accept` route
	* Added checkSeatOwnership function calls for both initiator and bidder
	* Checks if both the bidding initiator and accepted bidder still own their seats
	* Prevents invalid seat swaps due to outdated ownership information
3. Optimized bidding cleanup after seat swaps
	* Removes related bids and biddings for swapped seats
	* Updates other active biddings to remove invalid bids
4. Added new 'biddings-removed' event emission
	* Informs clients about removed biddings and affected seat numbers
	* Helps maintain consistency across all connected clients

### Testing
https://www.loom.com/share/8a75602c7b8c43989808902e45fd4e02

### CODE REVIEW

File: backend/routes/seatbid.js [hunk 1]
```diff
@@ -121,8 +135,25 @@ router.post('/accept', authenticateUser, async (req, res) => {
...
-        activeBiddings.delete(biddingId);
-        bids.delete(biddingId);
+        seatsToRemove.forEach(seatNumber => {
+            const biddingIdToRemove = `${bidding.flightId}-${seatNumber}`;
+            activeBiddings.delete(biddingIdToRemove);
+            bids.delete(biddingIdToRemove);
+        });
+
+        activeBiddings.forEach((activeBidding, activeBiddingId) => {
+            if (!seatsToRemove.includes(activeBidding.seatNumber)) {
```
Comment (alantulais) [inline backend/routes/seatbid.js:145]: @nowshin-108 Could you explain this logic? If I understand correctly, we are removing bids associated to the seat that was just accepted, right?

Quick question, do we allow users creating a bidding to also bid in other biddings?

- If not, can we do that?
- If yes, shall we invalidate their biddings if their seat was accepted in a separate bidding?

Feel free to implement this in separate PR.

File: backend/routes/seatbid.js [hunk 2]
```diff
@@ -121,8 +135,25 @@ router.post('/accept', authenticateUser, async (req, res) => {
             }),
         ]);
 
-        activeBiddings.delete(biddingId);
-        bids.delete(biddingId);
+        seatsToRemove.forEach(seatNumber => {
```
Comment (alantulais) [inline backend/routes/seatbid.js:138]: @nowshin-108 Just to confirm. This is to delete other bids associated to this seat if it was accepted in a bidding, right?

**Review by alantulais — APPROVED** [review]

File: backend/routes/seatbid.js (hunk 2 above)
Comment (nowshin-108) [reply backend/routes/seatbid.js:138]: Yeap!

File: backend/routes/seatbid.js (hunk 1 above)
Comment (nowshin-108) [reply backend/routes/seatbid.js:145]: Yeap, users do can create a bidding for their seat and also bid on other seats simultaneously. If a user's bid is accepted in a separate bidding, their original bidding and all their bids on other seats are automatically invalidated. 

`        seatsToRemove.forEach(seatNumber => {
            const biddingIdToRemove = `${bidding.flightId}-${seatNumber}`;
            activeBiddings.delete(biddingIdToRemove);
            bids.delete(biddingIdToRemove);
        });

        activeBiddings.forEach((activeBidding, activeBiddingId) => {
            if (!seatsToRemove.includes(activeBidding.seatNumber)) {
                const updatedBids = bids.get(activeBiddingId).filter(bid => 
                    !seatsToRemove.includes(bid.bidderSeatNumber)
                );
                bids.set(activeBiddingId, updatedBids);

                if (updatedBids.length === 0) {
                    activeBiddings.delete(activeBiddingId);
                    bids.delete(activeBiddingId);
                }
            }
        });`                                                                                                                       

This code basically removes the biddings and bids for the seats directly involved in the swap. Then, it iterates through all other active biddings, removing any bids made from the swapped seats and deleting biddings that end up with no bids. This is to ensure system-wide consistency after a seat swap.

**Pull request merged**